lib/perso.ml

```diff
@@ -2022,6 +2026,50 @@ and eval_item_field_var ell = function
       with Failure _ -> raise Not_found)
   | _ -> raise Not_found
 
+and eval_title_field_var conf base env (_p, (cnt, name, title, places, dates))
+    _loc = function
+  | [ "is_first" ] ->
+      VVbool
+        (match get_env "first" env with Vbool x -> x | _ -> raise Not_found)
+  | [ "is_main" ] ->
+      VVbool (match name with Tmain | Tnone -> true | Tname _x -> false)
+  | [ "cnt" ] -> VVstring (string_of_int cnt)
+  | [ "name" ] -> (
+      match name with
+      | Tname n -> VVstring (sou base n |> escape_html :> string)
+      | _ -> VVstring "")
+  | [ "title" ] -> VVstring (sou base title |> escape_html :> string)
+  | [ "places" ] ->
+      VVstring
+        (List.fold_left
+           (fun acc pl ->
+             acc
+             ^ (if acc <> "" then ", " else "")
+             ^ (sou base pl |> escape_html :> string))
+           "" places)
+  | [ "dates" ] ->
+      let date_opt_to_string d =
+        match d with
+        | Some (Dgreg (dmy, _)) ->
+            (DateDisplay.string_of_dmy conf dmy :> string)
+        | Some (Dtext d) -> (d |> escape_html :> string)
+        | None -> ""
+      in
+      VVstring
+        (List.fold_left
+           (fun acc (d1, d2) ->
+             acc
+             ^ (if acc <> "" then ", " else "")
+             ^
+             let s =
+               Format.sprintf "%s - %s" (date_opt_to_string d1)
+                 (date_opt_to_string d2)
+             in
+             if s = " - " then "? - ?" else s
+             (* FIXME should it return ""  instead of "? - ?" *))
+           "" dates)
```

Comment:
maybe do the date to string conversion in a separate step to make it more readable, something like:
```ocaml
let dates = List.map (fun (d1,d2) ->  
   let s = Format.sprintf "%s - %s" (date_opt_to_string d1) (date_opt_to_string d2) in
   if s = " - " then "? - ?" else s )  dates in
String.concat ", " dates
```

Reply:
Thanks for the comment.
I updated places and dates accordingly.
I am not sure about the "?" !! It might be better to return "" rather than "? - ?"

Reply:
thanks,
I'm not sure too

Reply:
Hold on a minute. I am now convinced that the ? are superfluous!!